lib/types.js

```diff
@@ -1950,29 +1950,32 @@ ArrayType.prototype._check = function (val, flags, hook, path) {
 
 ArrayType.prototype._read = function (tap) {
   var items = this.itemsType;
-  var val = [];
+  var val;
   var n;
+  var i = 0
```

Comment:
Nit: can you keep the style consistent throughout? For example semi-colons and variable grouping (initialized ones first and uninitialized on a single line after).

```js
var items = this.itemsType;
var i = 0;
var n, val;
```

Reply:
fixed.
Side-note: you should probably setup prettier for the project. I was a bit struggling with it because vscode kept re-formatting the text by default.